packages/dart/test/hub_span_test.dart

```diff
@@ -0,0 +1,311 @@
+
+          span2.end();
+          expect(hub.scope.getActiveSpan(), equals(span1));
+
+          span1.end();
+          expect(hub.scope.getActiveSpan(), isNull);
+        });
+
+        test('ending spans out of order removes them from active spans', () {
+          final hub = fixture.getSut();
+
+          final parent = hub.startSpan('parent');
+          final child = hub.startSpan('child');
+
+          parent.end();
+          expect(hub.scope.activeSpans, contains(child));
+          expect(hub.scope.activeSpans, isNot(contains(parent)));
+
+          child.end();
+          expect(hub.scope.activeSpans, isEmpty);
+        });
+
+        test(
+            'new span parents to active root span when multiple root spans exist',
+            () {
+          final hub = fixture.getSut();
+
+          final activeRoot =
+              hub.startSpan('active-root', parentSpan: null, active: true);
+          final inactiveRoot =
+              hub.startSpan('inactive-root', parentSpan: null, active: false);
+
+          final childToActiveSpan = hub.startSpan('child');
+          expect(childToActiveSpan.parentSpan, equals(activeRoot));
+          expect(childToActiveSpan.parentSpan, isNot(equals(inactiveRoot)));
+
+          final childToInactiveSpan =
+              hub.startSpan('child', parentSpan: inactiveRoot);
+          expect(childToInactiveSpan.parentSpan, equals(inactiveRoot));
+          expect(childToInactiveSpan.parentSpan, isNot(equals(activeRoot)));
+        });
+
+        test('deep hierarchy maintains correct parent chain', () {
+          final hub = fixture.getSut();
+
+          final spans = <Span>[];
+          for (var i = 0; i < 5; i++) {
+            spans.add(hub.startSpan('span-$i'));
+          }
+
+          // Verify chain: span0 <- span1 <- span2 <- span3 <- span4
+          expect(spans[0].parentSpan, isNull);
+          for (var i = 1; i < spans.length; i++) {
+            expect(spans[i].parentSpan, equals(spans[i - 1]));
+          }
+        });
+      });
+    });
+
+    group('captureSpan', () {
+      // TODO: add test that it was added to buffer
+
+      test('removes span from active spans on scope', () {
+        final hub = fixture.getSut();
+        final span = hub.startSpan('test-span');
+        expect(hub.scope.activeSpans, contains(span));
+
+        hub.captureSpan(span);
+
+        expect(hub.scope.activeSpans, isNot(contains(span)));
+      });
+
+      test('does nothing when hub is closed', () async {
+        final hub = fixture.getSut();
+        final span = hub.startSpan('test-span');
+        await hub.close();
+
+        // Should not throw
+        hub.captureSpan(span);
+      });
+
+      test('does nothing when tracing is disabled', () {
+        final hub = fixture.getSut(tracesSampleRate: null);
+
+        final span = hub.startSpan('test-span');
+        expect(span, isA<NoOpSpan>());
+        expect(hub.scope.activeSpans, isEmpty);
+
+        hub.captureSpan(span);
```

Comment:
Isn't maybe removeActiveSpan a more fitting name? It would also neatly mirror `setActiveSpan`